xkafka/internal/kafka.go

```diff
@@ -0,0 +1,26 @@
+// Package internal provides the kafka client interfaces for mocking.
+package internal
+
+import (
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+)
+
+// ConsumerClient is the interface for the confluent-kafka-go/kafka.Consumer.
+type ConsumerClient interface {
```

Comment:
internal package still feels unnecessary to me

My proposal is to make it package private interface for mocks and add a minimal interface for Consumer.

```go
// used in places like type definitions of ConsumerFunc
type Consumer interface {
	SubscribeTopics(topics []string, options ...ConsumerOption) error
	Unsubscribe() error
}

// used for mocking
type consumerClient interface {
	GetMetadata(topic *string, allTopics bool, timeoutMs int) (*kafka.Metadata, error)
	SubscribeTopics(topics []string, rebalanceCb kafka.RebalanceCb) error
	Unsubscribe() error
	Close() error
	readMessage(timeout time.Duration) (*kafka.Message, error)
}
```

Reply:
Similarly we can improve the API for publisher also.